tensorflow/contrib/learn/python/learn/datasets/__init__.py

```diff
@@ -64,3 +75,33 @@ def load_dataset(name, size='small', test_with_fake_data=False):
     return DATASETS[name](size, test_with_fake_data)
   else:
     return DATASETS[name]()
+
+def make_synthetic(name='circles', n_samples=100, noise=None, *args, **kwargs):
+  """Creates binary synthetic datasets
+
+  Args:
+    name: Name of the dataset to generate (optional, str, default='circles')
+    n_samples: Number of datapoints to generate (optional, int, default=100)
+    noise: Standard deviation of the Gaussian noise added (optional, float or None, default=None)
+
+  Returns:
+    Shuffled features and labels for given synthetic dataset of type `base.Dataset`
+
+  Raises:
+    ValueError: 
+
+  Note: 
+    - This is a generic synthetic data generator - individual generators might have more parameters!
+      See documentation for individual parameters
+    - Note that the `noise` parameter uses `numpy.random.normal` and depends on `numpy`'s seed
+
+  TODO:
+    - Support custom seed for replicability
+    - Support multiclass datasets
+    - Need shuffling routine. Currently synthetic datasets are reshuffled to avoid train/test correlation,
```

Comment:
nothing to change, but tf.learn automatically shuffle training data when fit is called.